assets/src/components/block-preview-label.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * WordPress dependencies
+ */
+import { Fragment } from '@wordpress/element';
+import { BlockIcon } from '@wordpress/block-editor';
+
+export default ( { label: name, block, blockType, displayIcon = true, alignIcon = 'left', accessibilityText } ) => {
+	if ( ! block ) {
+		return name;
+	}
+
+	let label;
+	let content;
+
+	// Todo: Cover more special cases if needed.
+	switch ( block.name ) {
+		case 'core/image':
+			if ( block.attributes.url ) {
+				content = block.attributes.url.slice( block.attributes.url.lastIndexOf( '/' ) ).slice( 0, 30 );
+
+				if ( content.length > 0 ) {
+					label = content;
+				}
+			}
```

Comment:
Maybe would be good to add `else` and set the label to `title` within it? Otherwise `label.length` might cause error below if the URL is not set.

Reply:
Thanks! 38b3452e8e05fab3bdd5a7761472cf5e23201127 addresses this by always setting a label already at the beginning.